Framework/Core/src/runDataProcessing.cxx

```diff
@@ -1611,6 +1611,91 @@ int runStateMachine(DataProcessorSpecs const& workflow,
           }
 
           // This should expand nodes so that we can build a consistent DAG.
+
+          // This updates the options in the runningWorkflow.devices
+          for (auto& device : runningWorkflow.devices) {
+            // ignore internal devices
+            if (device.name.find("internal") != std::string::npos) {
```

Comment:
`rfind("internal", 0)`, no?

Reply:
I do not think we have cases where a device would have "internal" more than once in its name. We can probably simplify this, by adding a bool flag to the DeviceSpec.